plugin/src/main/scala/org/virtuslab/akkasaferserializer/SaferSerializerPluginComponent.scala

```diff
@@ -0,0 +1,64 @@
+package org.virtuslab.akkasaferserializer
+
+import akka.actor.typed.Behavior
+
+import scala.tools.nsc.plugins.PluginComponent
+import scala.tools.nsc.{Global, Phase}
+
+class SaferSerializerPluginComponent(val pluginOptions: PluginOptions, val global: Global) extends PluginComponent {
+  import global._
+  override val phaseName: String = "akka-safer-serializer-gather"
+  override val runsAfter: List[String] = List("refchecks")
+
+  var rootsCache: List[Type] = List()
+
+  override def newPhase(prev: Phase): Phase =
+    new StdPhase(prev) {
+      override def apply(unit: global.CompilationUnit): Unit = {
+        val body = unit.body
+        val reporter = CrossVersionReporter(global)
+
+        val generics = Seq(
+          typeOf[akka.actor.typed.Behavior[_]],
+          typeOf[akka.persistence.typed.scaladsl.ReplyEffect[Any, _]],
```

Comment:
Would it be possible to make `akka-persistence` (same for `akka-projection`) hmmm an _optional dependency_? As in, we'd check `ReplyEffect` type params if `akka-persistence` is already on the classpath, but otherwise ignore it? 

Note that making it a _provided dependency_ is not the same: now if `akka-persistence` is **not** on the classpath, we'd likely get some ClassNotFound error.

Maybe relying on reflection would be a better idea here actually? so that you find `akka.persistence.typed.scaladsl.ReplyEffect` class by name?